modules/standard/Random.chpl

```diff
@@ -576,20 +733,38 @@ module Random {
       /*
         Returns the next value in the random stream.
 
-        Reals are between 0 and 1; integers are in the range of the integer.
-        :returns: The next value in the random stream as type :type:`eltType`.
+        Generated reals are in [0,1] - both 0.0 and 1.0 are
+        possible values.
+
+        Generated integers cover the full value range of the integer.
+
+        :arg resultType: the type of the result. Defaults to :type:`eltType`.
+          `resultType` must be the same or a smaller size number.
+        :returns: The next value in the random stream as type :type:`resultType`.
        */
-      proc getNext(): eltType {
+      proc getNext(type resultType=eltType): resultType {
```

Comment:
Yeah, I didn't like the feature when Ben requested it to me either (and said so).  I understand that it's "just bits" but it still seems goofy to me to say "This stream guarantees certain properties of distributions over values" and then to switch between different types of values mid-stream as there's no guarantee that the semantic values will have any properties w.r.t. the previous ones (so why use the same stream to generate them?).

I don't have a problem leaving it in for this release per se (especially given the date), but it's one of the things that makes me want to slap a big "this interface is almost certain to continue changing" labels across this module if we don't have one already.  It's also one of the things that irks me about the way this patch showed up for review -- to have a new routine injected into the code base sans any design discussion because one developer requested it seems very contrary to all of the "how to architect new features and avoid surprises / design by peer review" discussions we've had as a team over the past 6-12 months.

Philosophically, I agree that a serial random stream / number generator could/should have a completely different interface than a parallel one and that many of the "Python does this, so shouldn't Chapel?" features belong in that serial random number stream setting.